core/config/project_settings.cpp

```diff
@@ -700,7 +700,8 @@ Error ProjectSettings::_save_settings_binary(const String &p_file, const Map<Str
 	int count = 0;
 
 	for (Map<String, List<String>>::Element *E = props.front(); E; E = E->next()) {
-		for (List<String>::Element *F = E->get().front(); F; F = F->next()) {
+		for (String &F : E->get()) {
+			(void)F;
 			count++;
 		}
 	}
```

Comment:
This loop looks like something that could be removed (at least the inner one).

Reply:
@KoBeWi The problem is that List doesn't seem to have a method for getting the size.

Reply:
Ya, that should really be an addition to List<>.    Tracking the current size in list is really minimal overhead, and would bring List<> more inline with std::list

Reply:
Ok, but then what is the purpose of `(void)F;`?

Reply:
@KoBeWi `(void)F;` avoids a warning that the F variable is unused. This warning causes the compilation to fail since Godot treats all warnings as errors.

Reply:
So maybe add a comment about it.

Reply:
> The problem is that List doesn't seem to have a method for getting the size.

It has [`size()`](https://github.com/godotengine/godot/blob/master/core/templates/list.h#L466-L468) method. So that loop can be replaced with `count += E->get().size();`.

Reply:
^